### ISSUE TITLE
iOS notification appear when app is in foreground

### ISSUE THREAD

**Issue by RidClick** [andrehtissot/cordova-plugin-fcm-with-dependecy-updated#203]
I don't know if it's a problem with the plugin or the way I send the notification, or if I'm missing a parameter. But notifications on iOS appear when the app is in the foreground and background.
I would like them not to appear in the foreground.

Any suggestion?

Thanks in advance.

**Comment by inspire22** [comment]
Did you get this figured out at all?  This medium guide for this project implies it's supposed to be automatically silent while it is in the foreground.. https://medium.com/enappd/firebase-push-notifications-in-ionic-apps-d90c6550fdb7

**Comment by mvaljento** [comment]
I'm ran into the same issue the other day. So-called data messages (silent notifications) are working perfectly on Android, but I'm unable to get them working on iOS. They simply don't seem to arrive at all. I tried almost every example I could find online with different parameters, but whenever i remove the "notification" part of the Firebase payload, the iOS notifications stop working completely.

**Comment by RidClick** [comment]
Sorry for being late but I have not been able to make the notifications not show in background and foreground at the same time on iOS. As soon as I can solve it I will post it here.

**Comment by RidClick** [comment]
After researching for a couple of days I have found that my problem is that the APNs alert is always sent, it does not matter if I add the content-available property that is supposed to prevent the alert from being sent.

I still don't know if my problem is with the plugin or the Firebase Admin SDK which is the one I'm using to send notifications.
I am going to do a last test with the postman and the official url of the firebase but in advance I have sent a notification from the firebase itself to iOS and indeed, although the app is in foreground, the push notification is displayed and the APNs alert arrives.

I am more and more sure that the plugin is not controlling in iOS that the push notification does not show if it is already in the foreground.

I appreciate any suggestions.

**Comment by andrehtissot** [comment]
You are right Rid. The plugin doesn't present anything, it only receives and gives it to the JS context.

**Comment by RidClick** [comment]
It would be nice if you could add something to control it, since in my case I use a chat application with this plugin, on Android I have no problem but you imagine how it is a chat on iOS with all the notifications arriving while the app is open.
